src/main/resources/hub-detect-sh

```diff
@@ -30,8 +30,48 @@ DETECT_JAVA_OPTS=${DETECT_JAVA_OPTS:-}
 # DETECT_CURL_OPTS=--proxy http://myproxy:3128
 DETECT_CURL_OPTS=${DETECT_CURL_OPTS:-}
 
+PROXY_HOST=${BLACKDUCK_HUB_PROXY_HOST}
+PROXY_PORT=${BLACKDUCK_HUB_PROXY_PORT}
+PROXY_USERNAME=${BLACKDUCK_HUB_PROXY_USERNAME}
+PROXY_PASSWORD=${BLACKDUCK_HUB_PROXY_PASSWORD}
+
 SCRIPT_ARGS="$@"
 
+for i in "$SCRIPT_ARGS"
+do
+case $i in
+    --blackduck.hub.proxy.host=*)
+    PROXY_HOST="${i#*=}"
+    shift # past argument=value
+    ;;
+    --blackduck.hub.proxy.port=*)
+    PROXY_PORT="${i#*=}"
+    shift # past argument=value
+    ;;
+    --blackduck.hub.proxy.username=*)
+    PROXY_USERNAME="${i#*=}"
+    shift # past argument=value
+    ;;
+     --blackduck.hub.proxy.password=*)
+    PROXY_PASSWORD="${i#*=}"
+    shift # past argument=value
+    ;;
+    *)
+          # ignored option
+    ;;
+esac
+done
```

Comment:
The powershell does not pull it from the passed in args. Only from the environment.

Reply:
Do they get passed through to detect fine in the all caps, underscore? I assume they do because Spring? Powershell only looks for "blackduck.hub.url" but I can have it look for the all caps underscore one too.

Reply:
To pass environment variables to Detect they must be in all caps.  "blackduck.hub.url" would be passed in on the command line

Reply:
Ok, that's what I thought. I'll modify the powershell to match that. 
But I don't think you should also pull it from the script args. Force them to set it as an env variable.

Reply:
I think it is correct to encourage using an environment variable, but I think we're fine to accept all of them. It certainly can aid in testing to simply change my command line invocation instead of having to change the env var, source my profile, etc.

Reply:
But then we need to support it in the powershell as well and there is no way to do it easily and reliably. The powershell script will have to parse and escape itself. How will the scripts handle escaping, slashes, quotes? They will both need to handle them the same, so copying the args between bash and powershell works the same. I think it adds a world of trouble.